nomos-mix/message/src/lib.rs

```diff
@@ -62,3 +64,26 @@ pub fn unwrap_message(message: &[u8]) -> Result<(Vec<u8>, bool), Error> {
 pub fn is_drop_message(message: &[u8]) -> bool {
     message == DROP_MESSAGE
 }
+
+pub(crate) fn concat_bytes(bytes_list: &[&[u8]]) -> Vec<u8> {
+    let mut buf = Vec::with_capacity(bytes_list.iter().map(|bytes| bytes.len()).sum());
+    bytes_list
+        .iter()
+        .for_each(|bytes| buf.extend_from_slice(bytes));
+    buf
+}
```

Comment:
Would be better to:

```suggestion
#[inline]
pub(crate) fn concat_bytes(bytes_list: &[&[u8]]) -> Vec<u8> {
    bytes_list.into_iter().flat_map(|&s| s).copied().collect()
}
```

Reply:
Resolved by https://github.com/logos-co/nomos-node/pull/897#discussion_r1850026700